packages/ui/components/Button/index.tsx

```diff
@@ -3,42 +3,50 @@
 'use client';
 
 import React from 'react';
+import RenderIcon, { type IconProps } from '@mono/ui/components/RenderIcon';
 import styled, { css } from '@refract-ui/sc';
 import type { DefaultTheme } from 'styled-components';
 
-type colorProps = keyof DefaultTheme['themeColors'];
+type colorProps = keyof DefaultTheme['allColors'];
+type colorTokenProps = keyof DefaultTheme['colorTokens'];
 
 export type ButtonProps = {
-  color: colorProps;
+  $color: colorProps | colorTokenProps;
   size?: 'sm' | 'md' | 'lg';
   type?: 'button' | 'reset' | 'submit';
-  variant?: 'solid' | 'outlined' | 'link';
+  $variant?: 'rounded-outline' | 'link' | 'featured';
```

Comment:
@siobahnschuck 
I didn't port over any of the variant specific styles from since they seem specific to ECF, but I can scaffold some out if you think it would be helpful?